### PR TITLE
chore(backend): Move beneficiary to purchase

### PR DESCRIPTION
**Pull request by josipbagaric**
Task description: 

> Right now certificates expect to already have ONE fixed beneficiary when being uploaded. But this is not how the workflow with the on-chain certificates will go. ➝ Remove beneficiary if it is there
> 
> We need to allow multiple purchases per certificate. Add another field to the purchase which is amount. And then transfer & claim the on-chain certificate based on that. 
> 
> First the certificates will be extracted from the redemption statement and then we will add multiple claims with volumes and different beneficiaries to it.
> 
> In PL the purchases are basically equivalent to the claims and hold the functionality of adding multiple beneficiaries and volumes.   We should probably remove the beneficiary field from the certificates (or allow for multiple for the case when there are multiple claims). And add the stuff to the purchase entity & API that is needed for the on-chain claims (e.g. purpose).

### CODE REVIEW

File: apps/frontend/src/pages/ProofPage/effects.tsx [hunk 1]
```diff
@@ -41,9 +40,6 @@ export const useProductPageEffects = () => {
     generatorId: data?.certificate?.generatorId ?? '',
     energySource: <FuelType fuelType={data?.certificate?.energySource as FuelTypeEnum} />,
     region: `${data?.certificate?.country}, ${data?.certificate?.region}`,
-    redemptionDate: dayjs(data?.certificate?.redemptionDate).isValid()
```
Comment (alexworker23) [inline apps/frontend/src/pages/ProofPage/effects.tsx:44]: I think this should be replaced by `data?.redemptionDate` and not removed

File: apps/frontend/src/pages/ProofExamplePage/index.tsx [hunk 2]
```diff
@@ -107,7 +107,6 @@ const certificateInfoTableData: TableRowData<CertificateDto['id']>[] = [{
   id: exampleData.certificate.id ?? '',
   proofId: <EthereumAddress shortify clipboard address={exampleData.certificate.id ?? ''} />,
   product: exampleData.certificate.productType ?? '',
-  beneficiary: exampleData.certificate.beneficiary ?? '',
```
Comment (alexworker23) [inline apps/frontend/src/pages/ProofExamplePage/index.tsx:110]: I think this should be replaced by `exampleData.beneficiary ` and not removed

File: apps/frontend/src/pages/ProofPage/effects.tsx (hunk 1 above)
Comment (josipbagaric) [reply apps/frontend/src/pages/ProofPage/effects.tsx:44]: Totally yeah, forgot to change that. Thanks.

File: apps/frontend/src/pages/ProofExamplePage/index.tsx (hunk 2 above)
Comment (josipbagaric) [reply apps/frontend/src/pages/ProofExamplePage/index.tsx:110]: Totally yeah, forgot to change that. Thanks.

File: apps/frontend/src/pages/ProofPage/effects.tsx (hunk 1 above)
Comment (josipbagaric) [reply apps/frontend/src/pages/ProofPage/effects.tsx:44]: Done.

File: apps/frontend/src/pages/ProofExamplePage/index.tsx (hunk 2 above)
Comment (josipbagaric) [reply apps/frontend/src/pages/ProofExamplePage/index.tsx:110]: Done.

**Review by alexworker23 — APPROVED** [review]

**Pull request merged**